packages/@aws-cdk/core/test/arn.test.ts

```diff
@@ -165,20 +173,61 @@ nodeunitShim({
           resource: 'identitypool',
           resourceName: 'us-east-1:1a1a1a1a-ffff-1111-9999-12345678:bla',
           sep: '/',
+          arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+          // cannot be successfully parsed with CFN functions because of the colons in resourceName
+          checkCfnEncoding: false,
+        },
+        'arn:aws:servicecatalog:us-east-1:123456789012:/applications/0aqmvxvgmry0ecc4mjhwypun6i': {
+          resource: 'applications',
+          resourceName: '0aqmvxvgmry0ecc4mjhwypun6i',
+          sep: '/',
+          service: 'servicecatalog',
+          region: 'us-east-1',
+          account: '123456789012',
+          partition: 'aws',
+          arnFormat: ArnFormat.SLASH_RESOURCE_NAME_AND_RESOURCE,
         },
         'arn:aws:s3:::my_corporate_bucket': {
           partition: 'aws',
           service: 's3',
           region: '',
           account: '',
           resource: 'my_corporate_bucket',
+          arnFormat: ArnFormat.NO_RESOURCE_NAME,
+        },
+        'arn:aws:s3:::my_corporate_bucket/object.zip': {
+          partition: 'aws',
+          service: 's3',
+          region: '',
+          account: '',
+          resource: 'my_corporate_bucket/object.zip',
+          arnFormat: ArnFormat.NO_RESOURCE_NAME,
```

Comment:
This is definitely the most interesting test case. I can be convinced that it should return `arnFormat : ArnFormat.SLASH_RESOURCE_NAME`, `resource: 'my_corporate_bucket'`, `sep: '/'`, and `resourceName: 'object.zip'` instead of what it's returning right now.